org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -385,6 +390,40 @@ private static boolean isBinary(Path file) {
 				&& !fileName.endsWith(SOURCE_JAR_SUFFIX));
 	}
 
+	public static IPath resolveGlobPath(IPath base, String glob) {
+		IPath pattern = new org.eclipse.core.runtime.Path(glob);
+		IPath baseDir = pattern.isAbsolute() ? org.eclipse.core.runtime.Path.ROOT : base;
+		return baseDir.append(pattern); // Append cwd to relative path
+	}
+
+	private static Map<IPath, Set<String>> groupGlobsByPrefix(IPath base, Set<String> globs) {
```

Comment:
See https://github.com/eclipse/eclipse.jdt.ls/pull/1305#discussion_r357910532 for updated illustration